src/components/input/Button.tsx

```diff
@@ -32,45 +60,66 @@ const Button = function Button({
   expanded,
   pressed,
   title,
-
   icon: Icon,
   size = 'md',
   variant = 'secondary',
+  unstyled = false,
 
+  role,
   ...htmlAttributes
 }: ButtonProps) {
+  const styled = !unstyled;
+  const themed = styled && variant !== 'custom';
+  const sized = styled && size !== 'custom';
+
+  const ariaProps: Record<string, unknown> = {
+    'aria-label': title,
+  };
+
+  // aria-pressed and aria-expanded are not allowed for buttons with
+  // an aria role of `tab`. Instead, the aria-selected attribute is expected.
+  if (role === 'tab') {
+    ariaProps['aria-selected'] = pressed;
+  } else {
+    ariaProps['aria-pressed'] = pressed;
+    ariaProps['aria-expanded'] = expanded;
+  }
+
   return (
-    <ButtonBase
+    <button
+      role={role ?? 'button'}
       data-component="Button"
+      // Setting a default `type` can prevent undesired form submissions in
+      // certain cases
+      type="button"
+      {...ariaProps}
       {...htmlAttributes}
-      classes={classnames(
-        'focus-visible-ring transition-colors whitespace-nowrap flex items-center',
-        'font-semibold rounded-sm',
-        {
-          // Variants
+      className={classnames(
+        styled && {
+          'focus-visible-ring transition-colors whitespace-nowrap flex items-center':
+            true,
+        },
```

Comment:
Maybe this would be more clear as:

```suggestion
        {
          'focus-visible-ring transition-colors whitespace-nowrap flex items-center':
            styled,
        },
```